build-package.sh

```diff
@@ -67,6 +71,28 @@ function sourceScript () {
   fi
 }
 
+# If we have been asked to run within a Docker image, pull the image, then execute
+# this script within the container.
+#
+# Note: environment variables must be explicitly propagated to the container. The
+# ones we currently recognize are:
+#
+# - SWIFT_SNAPSHOT: the Swift toolchain to be used,
+# - KITURA_NIO: an optional compilation mode switch for Kitura,
+# - GCD_ASYNCH: an optional compilation mode switch for Kitura-net,
+# - TESTDB_NAME: the name of a database to be accessed during the build.
+#
+if [ -n "${DOCKER_IMAGE}" ]; then
+  echo ">> Executing build in Docker container: ${DOCKER_IMAGE}"
+  set -x
+  docker pull ${DOCKER_IMAGE}
```

Comment:
Do we need to pull? Won't `docker run` pull if it can't find the image?

Reply:
I think it will, but I think it's easier to see what the script does at a glance with the explicit pull.  (that might be partly my inexperience as a dockerer)